middleware.js

```diff
@@ -1,4 +1,38 @@
 var http = require('http');
 
 exports.createServer = function () {
-};
+  var middlewares = [];
+
+  function next(req, res, index) {
+    if(index == undefined) {
+      index = -1;
+    }
+    return function () {
+      for(var i = index + 1; i < middlewares.length; i++) {
```

Comment:
Hmm... I don't understand why you are using a FOR loop here.

If you're doing everything right, you won't need this FOR loop because usually in the client we will do:

```js
app.use('/path', function (req, res, next) {
  doSomething();
  next(); // or res.end('optional message');
});
```

Notice there's a call to `next()` which iterates to the next middleware, or `res.end()` which ends the iteration.